admin/services/class-indexable-post-provider.php

```diff
@@ -0,0 +1,115 @@
+<?php
+/**
+ * WPSEO plugin file.
+ *
+ * @package WPSEO\Admin\Services
+ */
+
+/**
+ * Represents the indexable post service.
+ */
+class WPSEO_Indexable_Service_Post_Provider implements WPSEO_Indexable_Service_Provider {
+
+	/**
+	 * Returns an array with data for the target object.
+	 *
+	 * @param integer $object_id The target object id.
+	 *
+	 * @return array The retrieved data.
+	 */
+	public function get( $object_id ) {
+
+		if ( ! $this->is_indexable( $object_id ) ) {
+			return array();
+		}
+
+		$meta_robots_adv = explode( ',', WPSEO_Meta::get_value( 'meta-robots-adv', $object_id ) );
+
+		$link_count = new WPSEO_Link_Column_Count();
+		$link_count->set( array( $object_id ) );
+
+		return array(
+			'object_id'                   => (int) $object_id,
+			'object_type'                 => 'post',
+			'object_subtype'              => get_post_type( $object_id ),
+			'permalink'                   => get_permalink( $object_id ),
+			'canonical'                   => $this->get_meta_value( 'canonical', $object_id ),
+			'title'                       => $this->get_meta_value( 'title', $object_id ),
+			'description'                 => $this->get_meta_value( 'metadesc', $object_id ),
+			'breadcrumb_title'            => $this->get_meta_value( 'bctitle', $object_id ),
+			'og_title'                    => $this->get_meta_value( 'opengraph-title', $object_id ),
+			'og_description'              => $this->get_meta_value( 'opengraph-description', $object_id ),
+			'og_image'                    => $this->get_meta_value( 'opengraph-image', $object_id ),
+			'twitter_title'               => $this->get_meta_value( 'twitter-title', $object_id ),
+			'twitter_description'         => $this->get_meta_value( 'twitter-description', $object_id ),
+			'twitter_image'               => $this->get_meta_value( 'twitter-image', $object_id ),
+			'is_robots_noindex'           => $this->translate_robots_noindex( $this->get_meta_value( 'meta-robots-noindex', $object_id ) ),
+			'is_robots_nofollow'          => $this->get_meta_value( 'meta-robots-nofollow', $object_id ) === '1',
+			'is_robots_noarchive'         => in_array( 'noarchive', $meta_robots_adv, true ),
```

Comment:
As this value is very specific, you might consider to just do a `strpos !== false` to check against the raw value from the Meta table.
This will save an `explode` and an `in_array` but also can handle additional spacing if present.